### ISSUE TITLE
Add local master rooms from TM3.

### ISSUE THREAD

**Issue by brazilofmux** [brazilofmux/tinymux#391]
Original [issue 388](https://code.google.com/p/tinymux/issues/detail?id=388) created by brazilofmux on 2007-09-23T02:03:17.000Z:

It's been awhile since I've surveyed zones, but TinyMUX has two kinds of
zoning: ZMO and PARENT ROOM.

A ZMO is a THING object. Other objects join this zone by @chzone themselves
to the thing. The ZMO controls access with an enter lock. Any player in the
zone is able to control other things in the same zone.  It allows a forms
of shared ownership. Likewise, $-commands placed on the ZMO are matched
whenever something in the zone executes a command. The contents of the ZMO
are ignored and can be anything.

A PARENT ROOM uses some of the same plumbing as a ZMO, but the concept is
very different. In this case, a set of rooms are @chzoned to a single room.
players and things cannot be @chzone to rooms. TinyMUX calls this parent
room a local master room, and indeed, it is something like a mini-master
room in that the exits in the parent room are available in all the other
rooms, and objects in the mini-master parent room are searched for commands
like objects in a master room. A player only need to be located in one of
these rooms to have access to the commands in the local master room.

Zones in TM3 implement these two forms, but also includes a hybrid
mechanism that is also called a local master room.  In this case, this
local master 'room' isn't actually a room. It's ZMO that has been set ZONE.
If the local_master_rooms configuration option is enabled, and the ZMO is
set ZONE, the ZMO's contents are also searched for $-commands.  In this
situation, the player does not need to be located in a zoned room, they are
themselves zoned to the ZMO.

There is a question about the need for the ZONE flag as it is performing
basically the same function as COMMANDS or !NO_COMMAND. Without ZONE, a ZMO
acts like a TinyMUX ZMO. With ZONE, a ZMO also searches its contents for
$-commands. There is some value in leaving the ZONE flag off the ZMO in th
einterest of prevents the ZMO's contents from matching $-commands whether
or not they are set NO_COMMAND.

Fortunately, this addition only touches a little code.
1. Add ZONE flag.
2. The local_master_rooms and local_master_parents configuration options
   only touch three places in command.cpp

The effect only opens up two new lines of $-command matching if enabled.
